### ISSUE TITLE
Add dropout

### ISSUE THREAD

**Issue by MollsReis** [EvilScott/brainy#2]
Randomly disable units (and their connections) during training to avoid overfitting

Paper: https://www.cs.toronto.edu/~hinton/absps/JMLRdropout.pdf
